test/examples/test_getting_started.py

```diff
@@ -0,0 +1,173 @@
+    roles = ['api']
+    app_replicaset_uuid = create_replicaset(admin_api_url, uris, roles)
+    wait_for_replicaset_is_healthy(admin_api_url, app_replicaset_uuid)
+
+    replicaset_roles = get_replicaset_roles(admin_api_url, app_replicaset_uuid)
+    # api role should contain vshard-router dependency
+    assert set(replicaset_roles) == set(['api', 'vshard-router'])
+
+    # create s1 replicaset
+    uris = [instances_conf[instance]['advertise_uri'] for instance in S1_INSTANCES]
+    roles = ['storage']
+    s1_replicaset_uuid = create_replicaset(admin_api_url, uris, roles)
+    wait_for_replicaset_is_healthy(admin_api_url, s1_replicaset_uuid)
+
+    replicaset_roles = get_replicaset_roles(admin_api_url, s1_replicaset_uuid)
+    # storage role should contain vshard-storage dependency
+    assert set(replicaset_roles) == set(['storage', 'vshard-storage'])
+
+    # create s2 replicaset
+    uris = [instances_conf[instance]['advertise_uri'] for instance in S2_INSTANCES]
+    roles = ['storage']
+    s2_replicaset_uuid = create_replicaset(admin_api_url, uris, roles)
+    wait_for_replicaset_is_healthy(admin_api_url, s2_replicaset_uuid)
+
+    replicaset_roles = get_replicaset_roles(admin_api_url, s2_replicaset_uuid)
+    # storage role should contain vshard-storage dependency
+    assert set(replicaset_roles) == set(['storage', 'vshard-storage'])
+
+    # bootstrap vshard
+    bootstrap_vshard(admin_api_url)
+
+    # test HTTP API
+    CUSTOMER_ID = 10
+    CUSTOMER_NAME = 'Elizabeth'
+    customer = {
+        'customer_id': CUSTOMER_ID,
+        'name': CUSTOMER_NAME
+    }
+
+    # create new customer
+    url = 'http://localhost:{}/storage/customers/create'.format(router_http_port)
+    r = requests.post(url, json=customer)
+    assert r.status_code == requests.status_codes.codes.CREATED
+    resp = r.json()
+    assert 'info' in resp
+    assert resp['info'] == 'Successfully created'
+
+    # # create the same customer again
+    # r = requests.post(url, json=customer)
+    # # XXX: r.status_code is 500 now
+
+    # get customer
+    url = 'http://localhost:{}/storage/customers/{}'.format(router_http_port, CUSTOMER_ID)
+    r = requests.get(url, json=customer)
+    assert r.status_code == requests.status_codes.codes.OK
+    resp = r.json()
+    assert resp == {
+        'customer_id': CUSTOMER_ID,
+        'name': CUSTOMER_NAME,
+        'accounts': [],
+    }
+
+    # get customer by wrong id
+    url = 'http://localhost:{}/storage/customers/{}'.format(router_http_port, CUSTOMER_ID+1)
+    r = requests.get(url, json=customer)
+    assert r.status_code == requests.status_codes.codes.NOT_FOUND
+    resp = r.json()
+    assert 'info' in resp
+    assert resp['info'] == 'Customer not found'
+
+    # update customer balance
+    # XXX: now I have no idea how to perform this call
+    # it requires account_id field, but now I don't even
+    # understand how to add an account to customer
+
+    # stop
+    cli.terminate()
```

Comment:
In case of failed test you will never terminate it. `request.addfinalizer(lambda: cli.terminate())`

Reply:
See `Cli` constructor.

Reply:
Well, then I don't understand why we should call cli.terminate twice?

Reply:
I've removed `cli.terminate` here, but generally I think that it's OK to call it twice (`terminate` defined in constructor just prevents running cli process after test fail, but here is the test emulating user session).